addons/website/static/src/snippets/s_instagram_page/options.js

```diff
@@ -0,0 +1,119 @@
+/** @odoo-module **/
+
+import {_t} from "@web/core/l10n/translation";
+import options from "@web_editor/js/editor/snippets.options";
+import SocialMediaOption from "@website/snippets/s_social_media/options";
+
+options.registry.InstagramPage = options.Class.extend({
+    /**
+     * @override
+     */
+    async onBuilt() {
+        const pageName = await this._getDbInstagramPageName();
```

Comment:
You call that method only once, do not make a function out of it (and no need to make it overriddable)

Reply:
Sad, I think it really improves the readability... Can I do a function that is declared inside the onBuilt function ?

Reply:
Randomly passing by.
By no mean do I think my opinion is the best one or that I am right, but here it is:

It's imo always a bad practice to have a "only call once" method to encapsulate some code.
It's also a bad practice to make a method which is only doing a single line operation even if called multiple times (unless that line is extremely difficult eg).

Otherwise, you enter the labyrinthe effect, you need to scroll down, scroll up, scroll down, scroll up, to read the code.
That's, IMHO, the first bad part.
The second bad part is that, the method name will add even more complexity unless you find the very perfect method name, which you won't because there is never one.
People will not get exactly the method name as you meant it to be, it will add a layer of confusion.

The same apply to variables, it's why when we can omit it, it's better.

This:
```py
def main():
   some_setup_basic_instruction()

   # Retrieve the XYZ
    instruction_1()
    instruction_2()
  
    # Adapt ABC
    instruction_1()
    while ():
        instruction_2()
    
    # Colour DEF
    instruction_1()
    if blabla:
        instruction_2()
```
Will imo always be better than
```py
def main():
   some_setup_basic_instruction()
   self.retrieve_xyz()
   self.adapt_abc()
   self.colour_def()

def retrieve_xyz()
    instruction_1()
    instruction_2()
  
def adapt_abc():
    instruction_1()
    while ():
        instruction_2()
    
def colour_def():
    instruction_1()
    if blabla:
        instruction_2()
```
Same for sub method
```py
def main():
    def retrieve_xyz()
        instruction_1()
        instruction_2()
  
    def adapt_abc():
        instruction_1()
        while ():
            instruction_2()
    
    def colour_def():
        instruction_1()
        if blabla:
            instruction_2()
          
   some_setup_basic_instruction()
   self.retrieve_xyz()
   self.adapt_abc()
   self.colour_def()
```

Because the first one just reads like a storyline, from A to Z, it's just like a simple flow.
And with good line return and comments, you can very well divide you code in clear sections, which is at the end what you are trying to accomplish by encapsulating it in variables.

It's exactly like the website generator PR, there was plenty of those "only called once" method or "short util" which made it impossible to read to me, I had to scroll to a sub method, to then scroll to another sub method, to then scroll to an util, to then scroll up the other way around.

Again, just my opinion.
But it's generally the first thing I ask to change.
`https://github.com/odoo/enterprise/pull/43608#discussion_r1298441773`
`https://github.com/odoo/odoo/pull/126719#discussion_r1322974772` (even if I was wrong and actually couldn't apply this in this case)

Reply:
> It's also a bad practice to make a method which is only doing a single line operation even if called multiple times (unless that line is extremely difficult eg).

I agree with this and with the fact that we have to prevent to have too much deep in the call stack but the rest of your opinion seems to be against clean code principles. I read everywhere that functions should have only one responsibility and be small (max 30 lines).

Reply:
I really disagree with that (as suggested by my previous message :innocent: ), maybe it's because Odoo already has well defined and well separated method, hooks etc.

I don't see the point to divide again a `start`, `onBuilt` js method or a `create`, `name_get` python method for example.

It's just that it turns into labyrinth nightmares, amplified by method and variable bad names (which are required when dividing into methods) most of the times, since if you "allow" that, people will overdo it.

Really just a few days ago:
- `https://github.com/odoo/odoo/pull/126719#discussion_r1321589230` this was making me have to scroll to see what `ref()` was, then go back the method, then the day after, I couldn't remember if it was an alias to `env.ref()` and/or if it was crashing in case of not found, has to scroll again
- `https://github.com/odoo/odoo/pull/126719#discussion_r1323135889` -> what does `_adjust_new_page_template` mean? You have to scroll to figure, the method name doesn't hint anything (it surely did to the writer but not to me)

I really don't think it brought added value to encapsulate that code.

The thing is that sadly, we have to go for "never" or "always" regarding this case, because people will overdo it every single time they can if you say something like "Let's not do it unless we judge it's really worth doing".
That's what "clean code principles" usually tell: "It's always a case by case, if you judge blabla, then do blabla, it should be obvious blabla, there is no 'always' or 'never', blabla".
But in real world, working with many people on many files, **you have to** instore a 'always' or "never". People don't even care about having 2 typos and 2 linter errors by line, they don't care about blaming line they are removing, they just won't care about making good usage of the method splitting, the easiest and fastest, the better. And thinking case by case is not the easiest so they won't.

But even if we did it (allow case by case), in this case, it would not be a good use case to split the method IMHO.
It's very very very short and the code which is left in `start()` is actually bound to the one encapsulated in  `_getDbInstagramPageName`. So you would actually need to move the code left in `start()` in the sub method which would be named something like `_setInstagramPage` which would get the info and set the attribute: encapsulate the whole behavior / principle. No?

Also, quickly googling, I read this from a random guy (so it's worth what it's worth) which is "quoting" a book:
> Also, please note that contrary to what many are saying here, **there appears to be ample evidence that moderately long functions (say 50 to 200 lines) actually have lower bug densities than lots of very short functions**. McConnell provides a bunch of references in his book (although I've seen those references disputed). Again, as stated above, there's always an element of judgment when determining what belongs in a function.

It might be worth considering too. Using variables (to hold submethod return value) (on top of other things) can surely lead to more bugs.

If you are strongly against it, I am sure @qsm-odoo will let it go if it's something you really care about / want.
For me, I was just passing by discussing this hot topic :smiley:

Reply:
We discussed it IRL, I will adapt my code to RDE/QSM comment :+1:

Reply:
Some notes to justify why I am asking that: I personally was more inclined to create one-call functions if it seemed to make sense before. Then I spent years now reviewing code... and not once did it make my life easier to have a function called only once. For the many reasons RDE explained:
- The name of the function was not clear to me
- I have to needlessly jump in the code
- etc

In Lou's PR for ChatGPT, there were like 10 occurences of those one-call functions... reuniting stuff was the only way for me to understand the uncommented code. Doing that also fixed bugs.

That's why I changed my mind over the years :wink: But if you prefer like you did before, I don't mind. Don't get me wrong: one-call functions can make sense to me. That's basically different interpretations of https://en.wikipedia.org/wiki/Single-responsibility_principle . It can also make sense for unit testing, possibility to override, etc. And I don't think coding principles are necessarily always right: e.g. I discovered OOP by myself as a teenager, studied it at school later, it was as much as a coding principle as a way of programming... and today, it's really not as popular as it was before from my point of view (OWL proving it, although it has some OOP aspects).